### ISSUE TITLE
Feature Request: Add choice of Bokeh's RadioButtonGroup to panel.ToggleButtons

### ISSUE THREAD

**Issue by DancingQuanta** [pyviz/panel#185]
Currently, the `pn.widgets.ToggleButtons` creates a Bokeh's `CheckboxButtonGroup` which allows a number of buttons in a group of buttons to be selected simultaneously.

I wanted to use Bokeh's 'RadioButtonGroup` instead, where I want only one button selected at a time.

Could this be acihieved through a keyword switch?

For example, to make a `RadioButtonGroup`
```
radio_buttons = pn.ToggleButtons(behaviour='radio')
```
and `behavour=checkbox` for a `CheckboxButtonGroup` and this would be the default as this was the first widget to be used as a `ToggleButtons`.

I will try a PR.

**Comment by xavArtley** [comment]
If I may, your remark raise some problems with class names of some widgets.
According to several article on internet and bokeh widgets (https://www.justinmind.com/blog/checkboxes-or-radio-buttons-let-the-ui-design-battle-commence/)

- CheckBoxes => any number of choices
- Radio => one and only one choice among possibilities

So for example RadioButton seems not the approriate name
![image](https://user-images.githubusercontent.com/18531147/50427863-87ace780-08b1-11e9-9e40-17505b3cacb7.png)

**Comment by philippjfr** [comment]
I agree the naming is not ideal and should probably be changed/fixed. Also for clarity, panel will never use ipywidgets, we explicitly moved away from ipywidgets after we initially built ``paramNB``.

**Comment by DancingQuanta** [comment]
Have there a written rationale for moving away from `ipywidgets`?

I could try to make a PR based on @xavArtley's idea which will make things a lot clearer.

**Comment by philippjfr** [comment]
The ability to seamlessly deploy apps in and outside the notebook is the main reason that panel exists. The other was that for years we had nothing but issues with the ipywidgets notebook extension, but that now seems to be resolved at least.

**Comment by xavArtley** [comment]
What do you think of  a specification like this?

Some definitions:
Toggle : `widget` which state active can be on or off (True|False).
CheckToggleGroup : Group of `widgets` which can be indepedently switched on or off
RadioToggleGroup : Group of `widgets` where only one is switched on

`Widgets` can be either buttons or boxes (advantage of boxes : distinction between check (square) and radio (round)

```
Toggle(name: String, active: Boolean)

CheckToggleGroup(objects: List, value: List)
RadioToggleGroup(objects: List, value: List(len 1) or Object) List for consistency?

class Toggle:
	display_type = ['button', 'box']

class ToggleGroup:
	display_type = ['button', 'box']
	mode = ['radio_group', 'check_group']
```

**Comment by xavArtley** [comment]
I made a notebook to illustrate a possible implementation of toggle groups
https://gist.github.com/xavArtley/3f5b0afe31bed7b1eb76a3846900aab4

**Comment by DancingQuanta** [comment]
Closed by #189
